hw/hw2-baseball.sql

```diff
@@ -0,0 +1,67 @@
+-- BDA_696 HW2 Script
+-- Each table will open with all players
+-- and then again for just one player (110029)
+-- for verification
+
+USE baseball;
+
+DROP TABLE IF EXISTS career_avg,
+	season_avg,
+	rolling_avg;
+    
+-- Create career average table
+
+CREATE TABLE career_avg
+	SELECT batter,
+		SUM(Hit) / SUM(atBat) AS bat_avg_career
+	FROM batter_counts WHERE atBat > 0
+	GROUP BY batter;
+
+SELECT *
+	FROM career_avg;
+	
+SELECT *
+	FROM career_avg WHERE batter = 110029;
+
+-- Create season average table
+
+CREATE TABLE season_avg
+	SELECT batter,
+		YEAR(game.local_date) as season,
+		SUM(Hit) / SUM(atBat) AS bat_avg_season
+	FROM batter_counts JOIN game ON batter_counts.game_id = game.game_id WHERE atBat > 0
+	GROUP BY batter, season
+	ORDER BY batter, season;
+
+SELECT *
+	FROM season_avg;
+	
+SELECT *
+	FROM season_avg WHERE batter = 110029;
+	
+-- Create 100 game rolling average
+
+CREATE TABLE rolling_avg
+	SELECT DATE(game.local_date) AS game_date,
+		batter,
+		(SUM(Hit)
+			OVER (PARTITION BY batter
+				ORDER BY game_date ASC ROWS BETWEEN 101 PRECEDING AND 1 PRECEDING)) 
+			/
+			(SUM(atBat)
+				OVER (PARTITION BY batter
+				ORDER BY game_date ASC ROWS BETWEEN 101 PRECEDING AND 1 PRECEDING))
+			AS avg_100_games_prior
+	FROM batter_counts JOIN game ON batter_counts.game_id = game.game_id WHERE atBat > 0
+	GROUP BY game_date ASC, batter
+	ORDER BY game_date ASC, batter;
+
```

Comment:
CREATE TABLE rolling_avg
	SELECT DATE(game.local_date) AS game_date, Hit, atBat,
		batter,
		(SUM(Hit)
			OVER (PARTITION BY batter
				ORDER BY game.local_date ASC ROWS BETWEEN 101 PRECEDING AND 1 PRECEDING)) 
			/
			(SUM(atBat)
				OVER (PARTITION BY batter
				ORDER BY game.local_date ASC ROWS BETWEEN 101 PRECEDING AND 1 PRECEDING))
			AS avg_100_games_prior
	FROM batter_counts JOIN game ON batter_counts.game_id = game.game_id WHERE atBat > 0
	GROUP BY game_date, game.local_date, Hit, atBat, batter
	ORDER BY game_date ASC, batter;

I was having issues running this third table in mysql. Here are the edits I made to make it work. A lot of the issues had to do with mysql having an only full group by mode (https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html)

Reply:
Looking into this now. It works with MariaDB but I know Jim is using MySQL also and having issues.

Reply:
Colin seems to be correct that it has to do with the differences between mysql and mariadb, this code Colin provided will run

Reply:
I can't really be sure what works on MySQL vs MariaDB since I'm only using Maria, so I'm just going to commit @colinanderson10 's changes. 

Thanks for the info on full group by, that's helpful to know. Seems really dumb that one would need to `select` and duplicate columns (like Hit and atBat) in a new table in MySQL when you can aggregate them in the `select` in Maria, but good to know.

Reply:
Yeah it seems quite redundant. There may be a better way to do it besides selecting the columns and grouping by them when you don't even want them in the final table, but that's what worked for me.